test/Microsoft.AspNet.Mvc.DataAnnotations.Test/CompareAttributeAdapterTest.cs

```diff
@@ -20,7 +21,17 @@ public void ClientRulesWithCompareAttribute_ErrorMessageUsesDisplayName()
             var metadata = metadataProvider.GetMetadataForProperty(typeof(PropertyDisplayNameModel), "MyProperty");
```

Comment:
This is now a new test. Leave the old one alone and add another with the non-`null` `IStringLocalizer` as you've done here. When done, all `*AttributeAdapterTest` classes should contain two tests showing the error message is correct with and without an `IStringLocalizer`.